services/provider/server/server.go

```diff
@@ -799,12 +806,27 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to update lastHeartbeat payload in the storageConsumer resource: %v", err)
 	}
 
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to get storageConsumer resource: %v", err)
+	}
+
 	channelName, err := s.getOCSSubscriptionChannel(ctx)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
 	}
 
-	return &pb.ReportStatusResponse{DesiredClientOperatorChannel: channelName}, nil
+	desiredClientConfigHash := getDesiredClientConfigHash(channelName, storageConsumer)
+
+	return &pb.ReportStatusResponse{DesiredClientOperatorChannel: channelName, DesiredConfigHash: desiredClientConfigHash}, nil
+}
+
+func getDesiredClientConfigHash(channelName string, storageConsumer *ocsv1alpha1.StorageConsumer) string {
+	var arr = []any{
+		channelName,
+		storageConsumer.Spec.StorageQuotaInGiB,
+	}
+	return util.GetMD5Hash(fmt.Sprintf("%v", arr))
```

Comment:
using `Sprintf` with the `%v` uses space as a delimiter, this creates the following serialization bug:

both `[]any{1, "ab cd", true}` and `[]any{"1 ab cd true"}` are getting serialized to the exact same string `[a ab true]`

Reply:
good catch, Done the change